### PR TITLE
fix(mcp): seed department/designation labels from the masters tenant_bootstrap copies (#1590)

### LINKED ISSUE #1590: [Bug] Add localization for one department value in dashboard
# Description
PMC_ELEC and PMC_PWD

Image

Image

### PR DESCRIPTION
**Pull request by KDwevedi**
Fixes #1590.

## What QA saw

The `ke` root dashboard's **Complaints by departments** tile renders two bars as raw codes — `PMC_ELEC`, `PMC_PWD` — beside correctly-named ones.

## Root cause (traced on bomet, not inferred)

| check | result |
|---|---|
| `complaint_facts` rows carrying `PMC_*` | 7, all under sub-tenant **`ke.india`** (root dashboard aggregates its sub-tenants) |
| `common-masters.Department` @ `ke.india` | all 8 PMC records present, each with a proper `name` |
| `COMMON_MASTERS_DEPARTMENT_PMC_*` messages | **zero, on every tenant on the platform** |
| `rainmaker-common` messages @ `ke.india` | 9, none of them a department |

`ke.india` was created by `tenant_bootstrap` from a canned dump (`pg.citya` / `india.citya` carry the identical 8 departments + 12 hierarchy rows). Step 2 copies the **masters** unconditionally; Step 6 copies **messages** as a union over source tenants — and can only carry a message that already exists somewhere. The source dump never localized those departments, so nothing existed to copy.

The dashboard then renders the code verbatim, correctly: `dimensionLabel` (`digit-ui-esbuild/products/dashboard/src/i18n/dimensionLabel.js`) resolves *locale message → data-owned name → raw code* and deliberately has no humaniser. This is the platform surfacing a real seeding gap, not a UI bug — so the fix belongs in the bootstrap layer.

## The fix

`deriveMasterLocalizations()` builds `COMMON_MASTERS_DEPARTMENT_<CODE>` / `COMMON_MASTERS_DESIGNATION_<CODE>` from each copied master's own `name`, applied in Step 6 as a **floor**:

- a real copied message always wins;
- a master with **no `name` yields nothing** — the gap stays visible rather than being papered over with a label invented from the code;
- **deployment default locale only** (`StateInfo.languages[0]`). The `name` is authored in whatever language the operator onboarded in, so it belongs in that pack; copying it into the others would defeat the locale-strict resolution from #1108/#1407 and hide genuine translation gaps;
- pushed to the **root tenant as well as the city**. `egov-localization` `_search` resolves a tenant's own rows only (verified on bomet: a search at `ke.india` returns its 9 rows and no `ke` row; a search at `ke` never sees a `ke.india` row). The employee shell loads packs for `stateTenantId`, so a city-only write would be invisible exactly where this was reported. Same shape as the existing `TENANT_TENANTS_*` push;
- the derived count is reported per locale in the tool result and as progress events — an operator sees the floor kick in.

## Tests

`digit-mcp/test-master-localizations.ts` — 17 assertions over the reported case, the key conventions, the no-fabrication rules, row-shape tolerance and dedupe. Wired into `digit-mcp-ci.yml`, which previously ran no unit tests at all (only the env-dependent suites exist, and those can't run in CI).

## Environment

bomet is already unblocked by hand: the 8 `COMMON_MASTERS_DEPARTMENT_PMC_*` messages are seeded at `ke` in `en_IN`, `pt_PT` and `fr_FR` (hand-translated, not English copies) and verified through the public `_search` path the dashboard uses. This PR stops the class of bug recurring on the next bootstrap.

## Noted, deliberately not in this PR

- `digit-mcp/src/utils/xlsx-loader.ts:840-847` sends the whole dept/desig/complaint-type localization batch inside `try { … } catch { /* Non-fatal */ }` and still reports `status: 'completed'` with `localization_keys: 0`. Different code path, same failure mode — worth its own one-concern PR.
- The `ke` root has substantial automated-test tenant pollution (`ke.pwtj*`, `ke.etoe*`, `ke.india`) which the root dashboard aggregates. That is a data-hygiene issue, not a localization one.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Automatically generates department and designation translations from master data.
  * Applies translations to primary and root tenant configurations while preserving existing translations.
  * Reports generated localization counts.

* **Bug Fixes**
  * Handles incomplete, malformed, duplicate, and inconsistently formatted master data reliably.
  * Treats master-data read failures as non-blocking.

* **Documentation**
  * Clarifies localization gap resolution and state-root tenant lookup and seeding guidance.

* **Tests**
  * Added offline coverage for localization generation and edge cases.
  * CI now runs the offline localization test suite.

### CODE REVIEW

File: digit-mcp/src/tools/mdms-tenant.ts [hunk 1]
```diff
@@ -2232,6 +2320,26 @@ export function registerMdmsTenantTools(registry: ToolRegistry): void {
...
+          // only, and only where the union has nothing — a real translation
+          // always wins over the master name.
+          let derivedCount = 0;
+          if (locale === primaryLocale) {
+            for (const m of derivedMasterMessages) {
+              const key = `${m.module}::${m.code}`;
+              if (byKey.has(key)) continue;
+              byKey.set(key, m);
+              derivedCount++;
+            }
```
[the remark this answers is not in the document]
Comment (KDwevedi) [reply digit-mcp/src/tools/mdms-tenant.ts:2340]: Checked this one against the code and I'm skipping it — the two StateInfos are the same record at the moment this runs.

Step 3's identity rewrite touches only `code` and `tenantId` on `common-masters.StateInfo` (`identityFieldsBySchema`, mdms-tenant.ts:1585). `languages` is copied from the source verbatim, and `tenant_bootstrap` takes no target-language argument — so on a fresh bootstrap `target.StateInfo.languages[0]` is by construction the same value as `source.StateInfo.languages[0]`. Reading the target would return an identical `primaryLocale` and add a round-trip.

The case where they genuinely diverge is a **re-run** against a target whose operator has since edited `StateInfo.languages` (bomet `ke` is exactly this — FRANÇAIS and PORTUGUÊS were added long after its bootstrap). But that divergence governs which locales the whole of Step 6 copies, not just the derived floor: `locales` drives the entire copy loop, so a target default missing from the source's list means that tenant gets *no pack at all* for it, derived labels or otherwise. Repointing locale discovery at the target is a behaviour change for every message this tool writes, which is more than this PR should carry.

Tracking it separately rather than folding it in here. Good catch on the other two — both fixed in 1a89f68f9.

**Review by vinothrallapalli-eGov — APPROVED** [review]
